File: HireHubDemo/HireHubDemo/hirehub-core/src/main/java/com/hirehub/core/entities/Candidate.java
package com.hirehub.core.entities;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "candidates")
@Data
public class Candidate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String name;

    private String skills; // JSON or Comma Separated
    private Double experience;

    @Column(name = "resume_url")
    private String resumeUrl;

    @Column(name = "profile_url")
    private String profileUrl;
}

File: HireHubDemo/hirehub-referral/Models/Candidate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HireHub.Referral.Models
{
    [Table("candidates")]
    public class Candidate
    {
        [Key]
        public long Id { get; set; }
        
        [Column("user_id")]
        public long UserId { get; set; }

        public string Name { get; set; }
        public string Skills { get; set; }
        
        // Navigation property
        public User User { get; set; }
    }
}
